vcd/resource_vcd_nsxt_distributed_firewall_rule.go

```diff
@@ -0,0 +1,395 @@
+	}
+	_, singleRule, err := vdcGroup.CreateDistributedFirewallRule(d.Get("above_rule_id").(string), firewallRuleType)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule create] error setting Distributed Firewall Rule: %s", err)
+	}
+
+	d.SetId(singleRule.Rule.ID)
+
+	return resourceVcdNsxtDistributedFirewallRuleRead(ctx, d, meta)
+}
+
+func resourceVcdNsxtDistributedFirewallRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentVdcGroup(d)
+	defer vcdClient.unlockParentVdcGroup(d)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] error retriving Org: %s", err)
+	}
+
+	vdcGroup, err := org.GetVdcGroupById(d.Get("vdc_group_id").(string))
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] error retrieving VDC Group: %s", err)
+	}
+
+	rule, err := vdcGroup.GetDistributedFirewallRuleById(d.Id())
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] error retrieving Firewall Rule By ID: %s", err)
+	}
+
+	firewallRuleType, err := getDistributedFirewallRuleType(vcdClient, d, "create")
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] error getting Distributed Firewall Rule type: %s", err)
+	}
+
+	firewallRuleType.ID = rule.Rule.ID
+	_, err = rule.Update(firewallRuleType)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule update] errorupdating Distributed Firewall Rule: %s", err)
+	}
+
+	return resourceVcdNsxtDistributedFirewallRuleRead(ctx, d, meta)
+}
+
+func resourceVcdNsxtDistributedFirewallRuleRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule read] error retriving Org: %s", err)
+	}
+
+	vdcGroup, err := org.GetVdcGroupById(d.Get("vdc_group_id").(string))
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule read] error retrieving VDC Group: %s", err)
+	}
+
+	rule, err := vdcGroup.GetDistributedFirewallRuleById(d.Id())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			d.SetId("")
+		}
+		return diag.Errorf("[Distributed Firewall Rule read] error retrieving Firewall Rule By ID: %s", err)
+	}
+
+	err = setDistributedFirewallRuleData(rule.Rule, d)
+	if err != nil {
+		return diag.Errorf("error storing data to state: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVcdNsxtDistributedFirewallRuleDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentVdcGroup(d)
+	defer vcdClient.unlockParentVdcGroup(d)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("[Distributed Firewall Rule delete] error retriving Org: %s", err)
```

Comment:
```suggestion
		return diag.Errorf("[Distributed Firewall Rule delete] error retrieving Org: %s", err)
```

Reply:
Actually, this came from other place (old DFW resource). I did a search and replace for all such occurrences.